src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionInfo.cs

```diff
@@ -0,0 +1,131 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Uno.UI.SourceGenerators.XamlGenerator.Utils
+{
+	internal sealed class XBindExpressionInfo
+	{
+		private readonly int _xBindCounter;
+		private readonly string _contextType;
+		private readonly string _contextName;
+
+		public XBindExpressionInfo(int xBindCounter, string contextType, string contextName)
+		{
+			_xBindCounter = xBindCounter;
+			_contextType = contextType;
+			_contextName = contextName;
+		}
+
+		/// <summary>
+		/// For an x:Bind that is semantically equivalent to `A.B?.C?.D.E`,
+		/// InstanceSubexpression1 is A.B?.C?.D (ie, stop on last null access).
+		/// This will be the whole expression if there are no nulls involved in the expression.
+		/// </summary>
+		/// <remarks>
+		/// InstanceSubexpression1 can include method names for invocations.
+		/// </remarks>
+		public string InstanceSubexpression1 { get; set; } = null!;
```

Comment:
Could it be `InstancePathExpression` and below `InstanceMemberExression` ?

Reply:
Well, the current names are indeed quite not very descriptive, but I also hardly understand the suggested names.

Would `ExpressionBeforeLastNullAccess` and `ExpressionAfterLastNullAccess` be good?

Reply:
That makes it more descriptive that numbers for sure. Let's go with that.